### PR TITLE
Enable SYCL atomic tests

### PR DESCRIPTION
**Pull request by masterleinad**
Depends on #3741.

### CODE REVIEW

File: core/unit_test/TestAtomicOperations_complexdouble.hpp [hunk 1]
```diff
@@ -45,6 +45,8 @@
 #include <TestAtomicOperations.hpp>
 
 namespace Test {
+// FIXME_SYCL atomic_fetch_oper for large types to be implemented
+#ifndef KOKKOS_ENABLE_SYCL
```
Comment (dalg24) [inline core/unit_test/TestAtomicOperations_complexdouble.hpp:49]: Why do you remove it from the list if everything is being commented out?

**Review by dalg24 — CHANGES_REQUESTED** [review]

File: core/unit_test/TestAtomicOperations_complexdouble.hpp (hunk 1 above)
Comment (masterleinad) [reply core/unit_test/TestAtomicOperations_complexdouble.hpp:49]: So that I have a better description for the FIXME.

Comment (masterleinad) [reply core/unit_test/TestAtomicOperations_complexdouble.hpp:49]: If you feel really strongly about it I can remove it but I prefer to have tests in `core/unit/CMakeLists.txt` that I have not looked at and found a better description.

Comment (crtrott) [reply core/unit_test/TestAtomicOperations_complexdouble.hpp:49]: I prefer tests which completely not work to be removed in cmake

Comment (dalg24) [reply core/unit_test/TestAtomicOperations_complexdouble.hpp:49]: You can have the comment in the CMakeLists.txt right next to the unit test you remove from the list

Comment (masterleinad) [reply core/unit_test/TestAtomicOperations_complexdouble.hpp]: OK. Changed.

**Review by dalg24 — APPROVED** [review]

**Comment by masterleinad** [comment]
Retest this please.

**Pull request merged**